[dominic2412] iP

dominic2412 opened the pull request:
**Skywalker Chatbot**
> "Do or do not, there is no try." – Yoda

Skywalker is a simple chatbot to help you manage your tasks. It's:

1. Text-based
2. Easy to use
3. Super fast
4. All you need to do:

Download the jar file from [here](https://github.com/dominic2412/ip/releases/tag/A-JUnit)

- Double-click it to start the program.
- Add tasks, deadlines, and more.
- And the best part is... it's FREE! 🎉

Features
- [x] Add tasks
- [x]  Mark tasks as done
- [ ] Find tasks by keyword
- [ ]  Recurring tasks (coming soon)


If you are a Java programmer, you can also use Skywalker to practice Java. Here’s the main method:

```
`public class Main {
    public static void main(String[] args) {
        Application.launch(SkywalkerApp.class, args);
    }
}`
```


Quick Usage Example:
Just type the following to add a task:
todo Read a book

ou can also use inline code for commands, like deadline and event. 

Complete the Duke chatbot (done!)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dominic2412 committed:
Level-0

dominic2412 committed:
Level-1

dominic2412 committed:
Level-2

dominic2412 committed:
Level-4

dominic2412 committed:
A-TextUiTesting

dominic2412 committed:
Level-5

dominic2412 committed:
Level-3

dominic2412 committed:
Level-6

dominic2412 committed:
attempt to Enum

dominic2412 committed:
branch-Level-7

dominic2412 committed:
branch-Level-8

dominic2412 committed:
no message

dominic2412 committed:
A-MoreOOP

dominic2412 committed:
A-Packages

dominic2412 committed:
A-Gradle

dominic2412 committed:
A-JUnit

dominic2412 committed:
Add find feature

dominic2412 committed:
Complete Java Doc

dominic2412 committed:
Check for coding standard

dominic2412 committed:
complete: fix Check style issues and ensure code compliance

dominic2412 committed:
New Feature: GUI

DKHB0207 commented on src/main/java/skywalker/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+     * @return The command object corresponding to the users' input
+     * @throws Exception If the command is unknown/invalid.
+     */
+    public static Command parse(String fullCommand) throws Exception {
```
Perhaps a more descriptive method name such as parseCommand would be better?

jerrelllee replied:
I think so too, since there is another method starting with parse as well

DKHB0207 commented on src/main/java/skywalker/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+            }
+            return todo;
+        case "D":
+            String by = parts[3];
```
Perhaps a more descriptive name such as completedBy would be better?

DKHB0207 reviewed (commented):
Hey Dominic! LGTM. You code looks great and seems to have little to no errors in terms of coding quality. Just some nits in terms of how descriptive your method and variable names. 

Overall, the code is well done and coding standard is well adhered to. Good job!

jerrelllee commented on src/main/java/skywalker/Skywalker.java:
```diff
@@ -0,0 +1,73 @@
+    /**
+     * Constructs a Skywalker instance with the specified file path for storing tasks.
+     *
+     * @param filePath the file path to the storage file
```
Please add punctuation behind each parameter description

jerrelllee commented on src/main/java/skywalker/command/MarkCommand.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    /**
+     * mark the task as done and prints a message
```
Please start your method header comments in the form "Marks ..."

jerrelllee commented on src/main/java/skywalker/exception/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,10 @@
+package skywalker.exception;
```
I like that you have well organised classes and packages

jerrelllee reviewed (commented):
Overall, I liked how you have well defined commands and exceptions that are customised for  the chatbot. I noted a few minor coding standard violations too.

dominic2412 committed:
Add: Assertion features

This ensures that invalid
states are caught early during development, reducing the likelihood of
bugs in production.

Assertions are added in the following places:
-Before adding a task to ensure the task object is not null
- When marking a task as completed to ensure the task exists in the list
- During deletion to verify the task index is within the valid range

These assertions make the code more robust by validating that critical
assumptions hold at runtime, improving code reliability.

dominic2412 committed:
Merge pull request #1 from dominic2412/branch-A-Assertions

Add: Assertion features

dominic2412 committed:
Apply Checkstyle fixes to all classes

Correct all Checkstyle violations across the project to ensure the code
follows the agreed style guidelines. These changes improve code readability
and maintain consistency throughout the codebase.

Key changes include:
- Reformatted code to adhere to line length limits
- Fixed missing Javadoc comments
- Adjusted spacing and indentation
- Resolved naming convention issues

These improvements help maintain clean, readable code and reduce potential
style-related errors during future development.

dominic2412 committed:
Merge pull request #2 from dominic2412/A-CodingStandard

Check for coding standard

dominic2412 committed:
Add duplicate detection feature
I
ntroduce functionality to detect and prevent duplicate tasks when
users attempt to add the same task multiple times.

Previously, the system allowed duplicate tasks, which could lead
to redundancy and confusion for the user. This change enhances
the user experience by checking for duplicate task descriptions
before adding a new task to the task list.

Changes made:
- Added a `isDuplicate()` method in TaskList to check if a task
  with the same description already exists.
- Modified `AddCommand` to prevent adding duplicate tasks and
  provide feedback to the user if a duplicate is detected.
- Updated `Ui` class to display a message when a duplicate task
  is detected.

dominic2412 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Apply Checkstyle fixes to all classes

dominic2412 committed:
Skywalker: Add unique personality and improve error handling

A-BetterGui:
- Updated the window panel to display the chatbot's name "Skywalker."
- Tweaked the chatbot's personality by customising its name and phrases.
- Enhanced the GUI with colours, icons, and a funny profile picture to represent Skywalker.

A-MoreErrorHandling:
- Improved error handling by adding customised exceptions for various anticipated errors.
- Ensured robustness when handling invalid user commands or system errors.

MoreTesting:
- Added more test cases to ensure thorough validation of commands and UI components.
- Increased code coverage for edge cases in the chatbot's functionality.

A-AiAssisted:
- Utilised AI tools like GitHub Copilot and ChatGPT to identify potential code improvements.
- Integrated Java Streams to make the code more efficient.

dominic2412 committed:
Update README.md

dominic2412 committed:
Update README.md

dominic2412 committed:
Update README to reflect Skywalker Chatbot enhancements

Update the README file to provide a more comprehensive overview of the
Skywalker Chatbot and the latest improvements added in the recent updates.

The changes made to the README include:
- Added detailed instructions on how to build and run the application.

This update helps new users better understand how to use the application.

dominic2412 committed:
Merge branch 'master' of https://github.com/dominic2412/ip

* 'master' of https://github.com/dominic2412/ip:
  Update README.md

# Conflicts:
#	docs/README.md

dominic2412 committed:
Update README.md

dominic2412 committed:
Update README.md

dominic2412 committed:
Update README.md

dominic2412 committed:
style: Fix code formatting issues and update Javadocs

- Fixed white space inconsistencies in the code.
- Added missing Javadocs for methods and classes.
- Improved the clarity and structure of existing Javadocs.
- Reformatted code to follow coding standards.

dominic2412 committed:
Update README.md

- Updated time format for event and deadline tasks to "yyyy-MM-dd'T'HHmm".
- Ensured all main commands (todo, deadline, event, list, mark, unmark, delete) work as expected.
- Completed testing for the latest version to verify functionality.

dominic2412 committed:
Update README.md
